### ISSUE TITLE
User conformers overwritten

### ISSUE THREAD

**Issue by lilyminium** [openforcefield/openff-interchange#346]
At present this code overwrites user-given conformers in-place:

https://github.com/openforcefield/openff-interchange/blob/a19ec32c65c9057a9c0ce81cc1a42931e6ea7ae4/openff/interchange/components/smirnoff.py#L304-L309

That seems like it could be an unpleasant surprise.

The below example comes after commenting out the if k_bondorder/length_bondorder part because I didn't have a handy force field on hand:

https://github.com/openforcefield/openff-interchange/blob/a19ec32c65c9057a9c0ce81cc1a42931e6ea7ae4/openff/interchange/components/smirnoff.py#L292-L302

```python
>>> from openff.toolkit.topology.molecule import Molecule, unit
Warning: Unable to load toolkit 'OpenEye Toolkit'. The Open Force Field Toolkit does not require the OpenEye Toolkits, and can use RDKit/AmberTools instead. However, if you have a valid license for the OpenEye Toolkits, consider installing them for faster performance and additional file format support: https://docs.eyesopen.com/toolkits/python/quickstart-python/linuxosx.html OpenEye offers free Toolkit licenses for academics: https://www.eyesopen.com/academic-licensing
>>> molecule = Molecule.from_smiles("CCO")
>>> molecule.generate_conformers(n_conformers=1)
>>> molecule.conformers[0] += 1 * unit.angstrom
>>> topology = molecule.to_topology()
>>> list(topology.reference_molecules)[0].conformers[0]
Quantity(value=array([[ 1.88165321,  0.95521882,  0.98525676],
       [ 0.41828996,  0.62427541,  1.05098497],
       [-0.35004062,  1.75806983,  1.17615782],
       [ 2.26504668,  1.17421359,  2.01224746],
       [ 2.01649295,  1.87054063,  0.39101094],
       [ 2.47635802,  0.10545035,  0.60814983],
       [ 0.21464441,  0.00317226,  1.96832828],
       [ 0.16449437, -0.00354494,  0.18411054],
       [-0.08693898,  2.51260405,  0.6237534 ]]), unit=angstrom)
>>>
>>> from openff.toolkit.typing.engines.smirnoff import ForceField
>>> sage = ForceField("openff-2.0.0.offxml")
>>>
>>> from openff.interchange.components.interchange import Interchange
>>> out = Interchange.from_smirnoff(force_field=sage, topology=topology)
>>> list(topology.reference_molecules)[0].conformers[0]
Quantity(value=array([[ 0.88165321, -0.04478118, -0.01474324],
       [-0.58171004, -0.37572459,  0.05098497],
       [-1.35004062,  0.75806983,  0.17615782],
       [ 1.26504668,  0.17421359,  1.01224746],
       [ 1.01649295,  0.87054063, -0.60898906],
       [ 1.47635802, -0.89454965, -0.39185017],
       [-0.78535559, -0.99682774,  0.96832828],
       [-0.83550563, -1.00354494, -0.81588946],
       [-1.08693898,  1.51260405, -0.3762466 ]]), unit=angstrom)
```

**Comment by mattwthompson** [comment]
Yeah, this implementation was the simplest/easiest. There's a too much complexity around `partial_bond_orders_from_molecules`/`charge_from_molecules` and how bond orders/conformers/partial charges are assigned in a force field. The SMIRNOFF spec offers pretty much no guidance on how to deal with cases in which either user-defined data or wrapped toolkit behavior are used, even though the toolkit implements a handful of these options.

Long-term these should be encoded into the spec. Short-term I suppose this should at least be documented, since this is a key behavior difference with the toolkit.

**Comment by lilyminium** [comment]
Would a deepcopy work for Interchange methods down the line? I just don't expect creating a new object Interchange from my existing object Topology to alter the existing object, especially without a warning.

**Comment by mattwthompson** [comment]
That's my intent - and [what was intended](https://github.com/openforcefield/openff-interchange/blob/a19ec32c65c9057a9c0ce81cc1a42931e6ea7ae4/openff/interchange/components/interchange.py#L182-L189) to be the current behavior. I can reproduce your code (after making L292 `if True or ( ...`), though. My guess is that the handlers should be passed the deepcopied `self.topology` instead of the argument `topology` as [they are now](https://github.com/openforcefield/openff-interchange/blob/a19ec32c65c9057a9c0ce81cc1a42931e6ea7ae4/openff/interchange/components/interchange.py#L226-L263). I'll give that spin.

**Comment by mattwthompson** [comment]
Uh, this is uncovering some spider webs.


```
ipdb> [b.bond.fractional_bond_order for b in topology.topology_bonds]
[None, None, None, None, None, None, None, None]
ipdb> [b.fractional_bond_order for b in [*topology.reference_molecules][0].bonds]
[0.98247731, 1.00093035, 0.97430863, 0.97548954, 0.97464897, 0.9529382, 0.95375943, 0.94134956]
```

I *don't* think this is related to https://github.com/openforcefield/openff-toolkit/issues/1147 since it goes through `TopologyBond.bond`, not `TopologyBond.molecule`. I can't reproduce this outside of a debugger that I placed inside of a failing test. Maybe it's time to dispose of this environment and make a new one.

**Comment by mattwthompson** [comment]
Thanks again for raising this issue - this should be fixed in the upcoming 0.2.0 release:


```python3
In [1]: >>> from openff.toolkit.topology.molecule import Molecule, unit
   ...: >>> molecule = Molecule.from_smiles("CCO")
   ...: >>> molecule.generate_conformers(n_conformers=1)
   ...: >>> molecule.conformers[0] += 1 * unit.angstrom
   ...: >>> topology = molecule.to_topology()
   ...: >>> list(topology.reference_molecules)[0].conformers[0]
Warning: Unable to load toolkit 'OpenEye Toolkit'. The Open Force Field Toolkit does not require the OpenEye Toolkits, and can use RDKit/AmberTools instead. However, if you have a valid license for the OpenEye Toolkits, consider installing them for faster performance and additional file format support: https://docs.eyesopen.com/toolkits/python/quickstart-python/linuxosx.html OpenEye offers free Toolkit licenses for academics: https://www.eyesopen.com/academic-licensing
Out[1]:
array([[ 1.88165321,  0.95521882,  0.98525676],
       [ 0.41828996,  0.62427541,  1.05098497],
       [-0.35004062,  1.75806983,  1.17615782],
       [ 2.26504668,  1.17421359,  2.01224746],
       [ 2.01649295,  1.87054063,  0.39101094],
       [ 2.47635802,  0.10545035,  0.60814983],
       [ 0.21464441,  0.00317226,  1.96832828],
       [ 0.16449437, -0.00354494,  0.18411054],
       [-0.08693898,  2.51260405,  0.6237534 ]]) <Unit('angstrom')>

In [2]: >>> from openff.toolkit.typing.engines.smirnoff import ForceField
   ...: >>> sage = ForceField("openff-2.0.0.offxml")
   ...: >>>
   ...: >>> from openff.interchange.components.interchange import Interchange
   ...: >>> out = Interchange.from_smirnoff(force_field=sage, topology=topology)
   ...: >>> list(topology.reference_molecules)[0].conformers[0]
Out[2]:
array([[ 1.88165321,  0.95521882,  0.98525676],
       [ 0.41828996,  0.62427541,  1.05098497],
       [-0.35004062,  1.75806983,  1.17615782],
       [ 2.26504668,  1.17421359,  2.01224746],
       [ 2.01649295,  1.87054063,  0.39101094],
       [ 2.47635802,  0.10545035,  0.60814983],
       [ 0.21464441,  0.00317226,  1.96832828],
       [ 0.16449437, -0.00354494,  0.18411054],
       [-0.08693898,  2.51260405,  0.6237534 ]]) <Unit('angstrom')>
